### DISCUSSION TITLE
Typechecking on `id` field when querying

### DISCUSSION THREAD

**Discussion by travis-cook-sfdc** [fastapi/sqlmodel#624] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import Field, Session, SQLModel, create_engine, select

class Hero(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    name: str


hero_1 = Hero(name="Foo")

engine = create_engine("sqlite://database.db")

SQLModel.metadata.create_all(engine)

with Session(engine) as session:
    session.add(hero_1)
    session.commit()

with Session(engine) as session:
    hero = session.exec(select(Hero)).one()
    x = dict[int, Hero] = {hero.id: hero}
```

### Description

* Create a Hero Model
* Save is to the database
* Query it from the database
* Run mypy
* See error:

```
scratch.py:20: error: Dict entry 0 has incompatible type "int | None": "Hero"; expected "int": "Hero"  [dict-item
```

Is there anyway for sqlmodel / mypy to resolve the fact that `id` can't be optional here?

### Operating System

macOS

### SQLModel Version

0.0.8

### Python Version

3.11

**Comment by YuriiMotov** [comment, marked as answer]
You can add `assert hero.id is not None` before that line.

I don't think there are other methods. Only checking the value in runtime
